### PR TITLE
fix(fraudserv): stop topics on Stop

### PR DESCRIPTION
**Pull request by Wondertan**
There was a blocker on the node side to implement this, but we have a solution for it now

### CODE REVIEW

**Review by renaynay — APPROVED** [review]
What's the solution? The fact we track topics and only try to close topics remaining in the map?

**Comment by Wondertan** [comment]
@renaynay, I will make a PR on celestia-node soon. It's the same solution as for FX update

File: fraudserv/service.go [hunk 1]
```diff
@@ -106,6 +106,10 @@ func (f *ProofService) Start(context.Context) error {
 // Stop removes the stream handler and cancels the underlying ProofService
 func (f *ProofService) Stop(context.Context) (err error) {
 	f.host.RemoveStreamHandler(protocolID(f.networkID))
+	for tp, topic := range f.topics {
+		delete(f.topics, tp)
```
Comment (vgonkivs) [inline fraudserv/service.go:110]: `f.topicsLk.Lock() `?

Comment (Wondertan) [reply fraudserv/service.go:111]: Good catch! Fixed

**Review by renaynay — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**